File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.py
# Definition for a binary tree node.
# class TreeNode:
#   def __init__(self, val=0, left=None, right=None):
#     self.val = val
#     self.left = left
#     self.right = right
class Solution:
  def sumNumbers(self, root: Optional[TreeNode]) -> int:
    ans = 0
    
    def dfs(node: Optional[TreeNode], curr: int) -> None:
      nonlocal ans
      if not node.left and not node.right:
        ans += curr+node.val
      if node.left:
        dfs(node.left, 10*(curr+node.val))
      if node.right:
        dfs(node.right, 10*(curr+node.val))
    
    dfs(root, 0)
    return ans

File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *   Val int
 *   Left *TreeNode
 *   Right *TreeNode
 * }
 */
func sumNumbers(root *TreeNode) int {
  ans := 0
  dfs(root, 0, &ans)
  return ans
}

func dfs(node *TreeNode, curr int, ans *int)  {
  if (*node).Left == nil && (*node).Right == nil {
    *ans += curr + (*node).Val
  }
  if (*node).Left != nil {
    dfs((*node).Left, 10 * (curr + (*node).Val), ans)
  }
  if (*node).Right != nil {
    dfs((*node).Right, 10 * (curr + (*node).Val), ans)
  }
}